File: tutorial/worked-examples/bac-replication-python/analysis/code/05_figures.py
# 05_figures.py - Publication figures

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Load event study coefficients
coef_hr = pd.read_csv(ANALYSIS / "output" / "tables" / "es_coefficients_hr.csv")
coef_nhr = pd.read_csv(ANALYSIS / "output" / "tables" / "es_coefficients_nhr.csv")

# Ensure output directory exists
(ANALYSIS / "output" / "figures").mkdir(parents=True, exist_ok=True)

# =============================================================================
# Figure 1: Hit-and-Run Event Study
# =============================================================================
fig, ax = plt.subplots(figsize=(10, 6))

coef_hr = coef_hr.sort_values('event_time')

# Plot with confidence bands
ax.fill_between(coef_hr['event_time'], coef_hr['ci_lower'], coef_hr['ci_upper'],
                alpha=0.3, color='steelblue')
ax.plot(coef_hr['event_time'], coef_hr['coefficient'], 'o-',
        color='steelblue', markersize=6, linewidth=2)

ax.axhline(y=0, color='gray', linestyle='--', linewidth=0.8)
ax.axvline(x=-0.5, color='red', linestyle='--', linewidth=0.8, alpha=0.7)

ax.set_xlabel('Years Since 0.08 BAC Law Adoption', fontsize=12)
ax.set_ylabel('Coefficient (log HR fatalities)', fontsize=12)
ax.set_title('Event Study: Hit-and-Run Fatalities', fontsize=14)

ax.spines['top'].set_visible(False)
ax.spines['right'].set_visible(False)
ax.grid(True, alpha=0.3)

plt.tight_layout()
plt.savefig(ANALYSIS / "output" / "figures" / "event_study_hr.png", dpi=300, bbox_inches='tight')
plt.close()

# =============================================================================
# Figure 2: Non-Hit-and-Run Event Study (placebo)
# =============================================================================
fig, ax = plt.subplots(figsize=(10, 6))

coef_nhr = coef_nhr.sort_values('event_time')

ax.fill_between(coef_nhr['event_time'], coef_nhr['ci_lower'], coef_nhr['ci_upper'],
                alpha=0.3, color='darkgreen')
ax.plot(coef_nhr['event_time'], coef_nhr['coefficient'], 'o-',
        color='darkgreen', markersize=6, linewidth=2)

ax.axhline(y=0, color='gray', linestyle='--', linewidth=0.8)
ax.axvline(x=-0.5, color='red', linestyle='--', linewidth=0.8, alpha=0.7)

ax.set_xlabel('Years Since 0.08 BAC Law Adoption', fontsize=12)
ax.set_ylabel('Coefficient (log non-HR fatalities)', fontsize=12)
ax.set_title('Event Study: Non-Hit-and-Run Fatalities (Placebo)', fontsize=14)

ax.spines['top'].set_visible(False)
ax.spines['right'].set_visible(False)
ax.grid(True, alpha=0.3)

plt.tight_layout()
plt.savefig(ANALYSIS / "output" / "figures" / "event_study_nhr.png", dpi=300, bbox_inches='tight')
plt.close()

# =============================================================================
# Figure 3: Combined Event Study (both outcomes)
# =============================================================================
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(14, 5))

# Hit-Run
ax1.fill_between(coef_hr['event_time'], coef_hr['ci_lower'], coef_hr['ci_upper'],
                 alpha=0.3, color='steelblue')
ax1.plot(coef_hr['event_time'], coef_hr['coefficient'], 'o-',
         color='steelblue', markersize=6, linewidth=2)
ax1.axhline(y=0, color='gray', linestyle='--', linewidth=0.8)
ax1.axvline(x=-0.5, color='red', linestyle='--', linewidth=0.8, alpha=0.7)
ax1.set_xlabel('Years Since Law Adoption', fontsize=11)
ax1.set_ylabel('Coefficient', fontsize=11)
ax1.set_title('(A) Hit-and-Run Fatalities', fontsize=12)
ax1.spines['top'].set_visible(False)
ax1.spines['right'].set_visible(False)
ax1.grid(True, alpha=0.3)

# Non-Hit-Run
ax2.fill_between(coef_nhr['event_time'], coef_nhr['ci_lower'], coef_nhr['ci_upper'],
                 alpha=0.3, color='darkgreen')
ax2.plot(coef_nhr['event_time'], coef_nhr['coefficient'], 'o-',
         color='darkgreen', markersize=6, linewidth=2)
ax2.axhline(y=0, color='gray', linestyle='--', linewidth=0.8)
ax2.axvline(x=-0.5, color='red', linestyle='--', linewidth=0.8, alpha=0.7)
ax2.set_xlabel('Years Since Law Adoption', fontsize=11)
ax2.set_ylabel('Coefficient', fontsize=11)
ax2.set_title('(B) Non-Hit-and-Run Fatalities', fontsize=12)
ax2.spines['top'].set_visible(False)
ax2.spines['right'].set_visible(False)
ax2.grid(True, alpha=0.3)

plt.tight_layout()
plt.savefig(ANALYSIS / "output" / "figures" / "event_study_combined.png", dpi=300, bbox_inches='tight')
plt.close()

print("  Created figures:")
print("    - event_study_hr.png")
print("    - event_study_nhr.png")
print("    - event_study_combined.png")

File: tutorial/worked-examples/bac-replication-r/analysis/code/05_figures.R
# 05_figures.R - Publication figures

library(ggplot2)
library(dplyr)
library(readr)

# Load event study coefficients
coef_hr <- read_csv(file.path(analysis, "output", "tables", "es_coefficients_hr.csv"),
                    show_col_types = FALSE)
coef_nhr <- read_csv(file.path(analysis, "output", "tables", "es_coefficients_nhr.csv"),
                     show_col_types = FALSE)

# Sort by event time
coef_hr <- coef_hr %>% arrange(event_time)
coef_nhr <- coef_nhr %>% arrange(event_time)

# =============================================================================
# Figure 1: Hit-and-Run Event Study
# =============================================================================
p_hr <- ggplot(coef_hr, aes(x = event_time, y = coefficient)) +
  geom_ribbon(aes(ymin = ci_lower, ymax = ci_upper), alpha = 0.3, fill = "steelblue") +
  geom_line(color = "steelblue", linewidth = 1) +
  geom_point(color = "steelblue", size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray50") +
  geom_vline(xintercept = -0.5, linetype = "dashed", color = "red", alpha = 0.7) +
  labs(
    title = "Event Study: Hit-and-Run Fatalities",
    x = "Years Since 0.08 BAC Law Adoption",
    y = "Coefficient (log HR fatalities)"
  ) +
  theme_minimal(base_size = 12) +
  theme(
    panel.grid.minor = element_blank(),
    plot.title = element_text(size = 14, face = "bold")
  )

ggsave(file.path(analysis, "output", "figures", "event_study_hr.png"),
       plot = p_hr, width = 10, height = 6, dpi = 300)

# =============================================================================
# Figure 2: Non-Hit-and-Run Event Study (placebo)
# =============================================================================
p_nhr <- ggplot(coef_nhr, aes(x = event_time, y = coefficient)) +
  geom_ribbon(aes(ymin = ci_lower, ymax = ci_upper), alpha = 0.3, fill = "darkgreen") +
  geom_line(color = "darkgreen", linewidth = 1) +
  geom_point(color = "darkgreen", size = 3) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray50") +
  geom_vline(xintercept = -0.5, linetype = "dashed", color = "red", alpha = 0.7) +
  labs(
    title = "Event Study: Non-Hit-and-Run Fatalities (Placebo)",
    x = "Years Since 0.08 BAC Law Adoption",
    y = "Coefficient (log non-HR fatalities)"
  ) +
  theme_minimal(base_size = 12) +
  theme(
    panel.grid.minor = element_blank(),
    plot.title = element_text(size = 14, face = "bold")
  )

ggsave(file.path(analysis, "output", "figures", "event_study_nhr.png"),
       plot = p_nhr, width = 10, height = 6, dpi = 300)

# =============================================================================
# Figure 3: Combined Event Study (both outcomes)
# =============================================================================
# Combine data
coef_hr$outcome <- "Hit-and-Run"
coef_nhr$outcome <- "Non-Hit-and-Run"
coef_combined <- bind_rows(coef_hr, coef_nhr)

p_combined <- ggplot(coef_combined, aes(x = event_time, y = coefficient, color = outcome, fill = outcome)) +
  geom_ribbon(aes(ymin = ci_lower, ymax = ci_upper), alpha = 0.2, color = NA) +
  geom_line(linewidth = 1) +
  geom_point(size = 2) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "gray50") +
  geom_vline(xintercept = -0.5, linetype = "dashed", color = "red", alpha = 0.7) +
  facet_wrap(~outcome, ncol = 2) +
  scale_color_manual(values = c("Hit-and-Run" = "steelblue", "Non-Hit-and-Run" = "darkgreen")) +
  scale_fill_manual(values = c("Hit-and-Run" = "steelblue", "Non-Hit-and-Run" = "darkgreen")) +
  labs(
    x = "Years Since Law Adoption",
    y = "Coefficient"
  ) +
  theme_minimal(base_size = 11) +
  theme(
    panel.grid.minor = element_blank(),
    legend.position = "none",
    strip.text = element_text(face = "bold")
  )

ggsave(file.path(analysis, "output", "figures", "event_study_combined.png"),
       plot = p_combined, width = 14, height = 5, dpi = 300)

cat("  Created figures:\n")
cat("    - event_study_hr.png\n")
cat("    - event_study_nhr.png\n")
cat("    - event_study_combined.png\n")
